torch/csrc/jit/ir.cpp

```diff
@@ -461,13 +461,107 @@ void Block::cloneFrom(Block * src, std::function<Value*(Value*)> value_map) {
   }
 }
 
+bool Value::equals(const Value* other) const {
+  return *this->type_ == *other->type_;
```

Comment:
Hmm considering all Values of the same type as equal seems like a bad idea, because they generally have many other fields like the offset into node's outputs. Do we really need to define some equality on them, or could we simply only compare nodes. It seems sufficient for me.